tests/CanadaPost/Tests/RatingTest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace CanadaPost\Tests;
+
+use CanadaPost\Rating;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Psr7\Response;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * Rating Class Tests.
+ * @group Rating
+ */
+class RatingTest extends PHPUnit_Framework_TestCase
+{
+
+    /**
+     * @var Rating $ratingService
+     */
+    protected $ratingService;
+
+    /**
+     * Build the $ratingService to be tested.
+     */
+    public function setUp()
+    {
+        $config = [
+            'username' => 'username',
+            'password' => 'password',
+            'customer_number' => 'customer_number',
+        ];
+        $this->ratingService = new Rating($config);
+    }
+
+    /**
+     * Test the POST request to get rates from Canada Post.
+     * @covers Rating::getRates()
+     * @test
+     */
+    public function getRates()
+    {
+        $body = file_get_contents(__DIR__ . '/../Mocks/rating-response-success.xml');
+        $mock = new MockHandler([
+            new Response(200, [], $body),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $response = $this->ratingService->getRates('', '', 1, ['handler' => $handler]);
+        foreach ($response['price-quotes'] as $quotes) {
+            foreach ($quotes as $rate) {
+                $this->assertTrue(is_array($rate));
+                $this->assertNotEmpty($rate['service-code']);
+                $this->assertNotEmpty($rate['service-name']);
+                $this->assertGreaterThan(0, $rate['price-details']['base']);
+            }
+        }
+    }
+
+}
```

Comment:
You need to be clear on what exactly you are testing here: that the response was properly parsed and the values are not altered in any way. Checking if `$rate` is an array (you could also check `price-quotes`) and checking if we do have values is ok, it checks that we have the expected structure. You do know however the exact expected values, you provided them, so check that they are equal to ensure that the values are not altered in any way while parsing the response.

Reply:
That means, for example, `$this->assertEquals('9.59', $rate['price-details']['base']);`

Reply:
to test all the price values i would have to only test one rate then.

or remove the foreach and test something like this:

`$this->assertEquals('9.59', $rates[0]['price-details']['base']);`
`$this->assertEquals('22.64', $rates[1]['price-details']['base']);`
`$this->assertEquals('9.59', $rates[2]['price-details']['base']);`
`$this->assertEquals('12.26', $rates[2]['price-details']['base']);`

is that what you want?